src/group_theory/group_action/option.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import group_theory.group_action.defs
+
+/-!
+# Option instances for additive and multiplicative actions
+
+This file defines instances for additive and multiplicative actions on `option` type. Scalar
+multiplication is defined by `a • some b = some (a • b)` and `a • none = none`.
+
+## See also
+
+* `group_theory.group_action.pi`
+* `group_theory.group_action.prod`
+* `group_theory.group_action.sigma`
+* `group_theory.group_action.sum`
+-/
+
+variables {M N P α β γ : Type*}
+
+namespace option
+
+section has_smul
+variables [has_smul α β] [has_smul α γ] [has_smul β γ] (a : α) (b : β) (x : option β)
+
+@[to_additive option.has_vadd] instance : has_smul α (option β) := ⟨λ a, option.map $ (•) a⟩
+
+@[to_additive] lemma smul_def : a • x = x.map ((•) a) := rfl
+@[simp, to_additive] lemma smul_none : a • (none : option β) = none := rfl
+@[simp, to_additive] lemma smul_some : a • some b = some (a • b) := rfl
+
+instance [is_scalar_tower α β γ] : is_scalar_tower α β (option γ) :=
+⟨λ a b x, by { cases x, exacts [rfl, congr_arg some (smul_assoc _ _ _)] }⟩
+
+@[to_additive] instance {α β γ : Type*} [has_smul α γ] [has_smul β γ] [smul_comm_class α β γ] :
```

Comment:
```suggestion
@[to_additive] instance [has_smul α γ] [has_smul β γ] [smul_comm_class α β γ] :
```

Reply:
No, this introduces a stray `has_smul α β`.